cfgov/ask_cfpb/models/answer_page.py

```diff
@@ -349,6 +350,30 @@ def __str__(self):
         else:
             return self.title
 
+    def validate_unique(self, *args, **kwargs):
+        super().validate_unique(*args, **kwargs)
+
+        if self.answer_base is None:
+            return
+
+        # Ensure that there is only ever one answer page in this language for
+        # an answer
+        pages_with_same_answer_and_language = self.__class__.objects.filter(
+            language=self.language,
+            answer_base=self.answer_base
+        ).exclude(
+            pk=self.pk
+        )
+        if pages_with_same_answer_and_language.exists():
+            raise ValidationError(
+                {
+                    NON_FIELD_ERRORS: (
+                        f"An answer page in {self.language} already exists "
```

Comment:
I briefly considered something like this to avoid the odd "For answer page in en" error messaging, but subsequent error catching could make this moot.

```python
from django.conf import settings
'...'
language_map = {tup[0]: f"{tup[1]}" for tup in settings.LANGUAGES}
'...'
f"An answer page in {language_map[self.language]} already exists"
```